### DISCUSSION TITLE
[New Feature] Allow FreeUnits to be spawned on their building's dock or pad

### DISCUSSION THREAD

**Discussion by MNOrac** [Vinifera-Developers/Vinifera#1441] [Feature Requests]
### Description:
In vanilla TS, freeunits spawn on an un-occupied cell bordering the structure they come with. It would be nice to be able to have them spawn on their building when the building has a dock. This would allow artists to include the unit in the buildup animation without guessing at where it will actually appear, and remove the need for players to task freeunits out of the way of further construction.

### Possible Implementation:
Since the goal of Vinifera is to leave existing behaviours untouched, an additional key on the building such as `FreeUnitStartsDocked=yes` or something less wordy seems appropriate.

**Comment by CCHyper** [comment]
While I'm happy that this should be easy to add an option for, I can see a possible issue if for some reason the area becomes blocked at the point the `FreeUnit` is spawned, which would visually look bugged.
